Week3/homework/step2-6.js

```diff
@@ -5,12 +5,24 @@ const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
   // Replace this comment and the next line with your code
-  console.log(arr);
+  const result = [];
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      result.push(arr[i][j]);
+    }
+  }
+  return result;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const result = [];
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++)
+      for (let y = 0; y < arr[i][j].length; y++) {
```

Comment:
Most developers use the letters i, j, k, m, n for `for` loop index variables, in this case `k` for the third loop. Note that the letter `l` should be avoided as it is difficult to discriminate it from the number `1`.